discretize/TreeMesh.py

```diff
@@ -724,10 +730,28 @@ def gridCC(self):
 
     @property
     def gridN(self):
+        """
+        Returns an M by N numpy array with the widths of all cells in order.
+        M is the number of nodes and N=1,2,3 is the dimension of the mesh.
+        """
+
         self.number()
         R = self._deflationMatrix('N', withHanging=False)
         return R.T * self._gridN + np.repeat([self.x0],self.nN,axis=0)
 
+    @property
+    def h_gridded(self):
+        """
+        Returns an (nC, dim) numpy array with the widths of all cells in order
+        """
+
+        if getattr(self, '_h_gridded', None) is None:
```

Comment:
Is this deleted when the mesh is changed?
I would need to look at the code to point you to exactly where that is done, but if the mesh is refined/coarsened then this needs to be deleted.

Reply:
nice catch, no this would still be stashed if it had been created

Reply:
It is/isn't stashed in the exact same way the gridCC is/isn't stashed. I'll leave that decision to the higher ups.

Reply:
@dccowan: this is the issue that @rowanc1 is referring to. You are using "lazy loading" here. The first time `h_gridded` is asked for, it is created and stored as a private variable (`_h_gridded`). It the mesh is refined, right now `_h_gridded` is not deleted

![image](https://user-images.githubusercontent.com/6361812/34009710-d0029b2c-e0be-11e7-8e71-ae8452d649b1.png)

Reply:
In the last commit, this is now cleaned up.